src/ray/core_worker/core_worker.cc

```diff
@@ -677,6 +685,63 @@ void CoreWorker::InternalHeartbeat(const boost::system::error_code &error) {
   internal_timer_.async_wait(boost::bind(&CoreWorker::InternalHeartbeat, this, _1));
 }
 
+void CoreWorker::LocationResolveHeartBeat(const boost::system::error_code &error) {
+  if (error == boost::asio::error::operation_aborted) {
+    return;
+  }
+  // Note: This will poll the GCS service every certain period.
+  // This exists to resolve scenarios such as
+  // https://github.com/ray-project/ray/pull/8679/files
+  ResolveActorsLocationNotPersistedToGCS();
+  internal_timer_.expires_at(
+      internal_timer_.expiry() +
+      boost::asio::chrono::milliseconds(kLocationResolveHeartbeatMillis));
+  internal_timer_.async_wait(
+      boost::bind(&CoreWorker::LocationResolveHeartBeat, this, _1));
+}
+
+void CoreWorker::ResolveActorsLocationNotPersistedToGCS() {
+  absl::MutexLock lock(&actor_location_resolve_waiters_mutex_);
+  for (auto it = actor_location_resolve_waiters_.begin();
+       it != actor_location_resolve_waiters_.end();) {
+    auto current = it++;
+    const auto &actor_id = *current;
+    ActorHandle *actor_handle = nullptr;
+    RAY_CHECK_OK(GetActorHandle(actor_id, &actor_handle));
+
+    if (actor_handle->IsPersistedToGCS()) {
+      // if an actor is already persisted to GCS, GCS will be responsible for reporting
+      // state of actors, so just remove it from the waiter.
+      actor_location_resolve_waiters_.erase(actor_id);
```

Comment:
It's not a good idea to erase while iterating through the vector. Instead, you can do `it++` if you don't want to erase and `it = actor_location_resolve_waiters_.erase(it)` if you want to erase.

Reply:
If you look up there, I used `auto current = it++`, which is used in other parts of codebase that is for removing entries during iteration, so I assume it is fine? Please correct me if I am wrong!

Reply:
Ah yes, I think that part is fine, but usually to erase while iterating, you should call `.erase(it)` instead of `.erase(key)`. Just realized that `flat_hash_set.erase(it)` returns void, though, so ignore my earlier comment!

Also, you can instead do `it++` in the for loop declaration, like:
`for (auto it = set.begin(); it != set.end(); it++)`